x-pack/plugin/eql/src/test/java/org/elasticsearch/xpack/eql/optimizer/TomlFoldTests.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.eql.optimizer;
+
+import com.carrotsearch.randomizedtesting.annotations.ParametersFactory;
+import org.elasticsearch.common.unit.TimeValue;
+import org.elasticsearch.tasks.TaskId;
+import org.elasticsearch.test.ESTestCase;
+import org.elasticsearch.xpack.eql.expression.function.EqlFunctionRegistry;
+import org.elasticsearch.xpack.eql.parser.EqlParser;
+import org.elasticsearch.xpack.eql.session.EqlConfiguration;
+import org.elasticsearch.xpack.ql.expression.Expression;
+import org.elasticsearch.xpack.ql.expression.function.Function;
+import org.elasticsearch.xpack.ql.expression.function.FunctionDefinition;
+import org.elasticsearch.xpack.ql.expression.function.UnresolvedFunction;
+
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static java.util.stream.Collectors.toList;
+
+public class TomlFoldTests extends ESTestCase {
+    protected static final String PARAM_FORMATTING = "%1$s.test -> %2$s";
+
+    @ParametersFactory(shuffle = false, argumentFormatting = PARAM_FORMATTING)
+    public static List<Object[]> readTestSpecs() throws Exception {
+        List<EqlFoldSpec> foldSpecs = EqlFoldSpecLoader.load("/test_folding.toml");
+        foldSpecs.addAll(EqlFoldSpecLoader.load("/test_string_functions.toml"));
+        List<EqlFoldSpec> unsupportedSpecs = EqlFoldSpecLoader.load("/test_unsupported.toml");
+
+        HashSet<EqlFoldSpec> filteredSpecs = new HashSet<>(foldSpecs);
+        filteredSpecs.removeAll(unsupportedSpecs);
+        return asArray(filteredSpecs);
+    }
+
+    public static List<Object[]> asArray(Collection<EqlFoldSpec> specs) {
+        AtomicInteger counter = new AtomicInteger();
+        return specs.stream().map(spec -> new Object[] {
+            counter.incrementAndGet(), spec
+        }).collect(toList());
+    }
+
+    private static final EqlConfiguration caseSensitiveConfig = config(true);
+    private static final EqlConfiguration caseInsensitiveConfig = config(false);
+    private static EqlParser parser = new EqlParser();
+    private static final EqlFunctionRegistry functionRegistry = new EqlFunctionRegistry();
+
+    private final int num;
+    private final EqlFoldSpec spec;
+
+    private static EqlConfiguration config(boolean isCaseSensitive) {
+        return new EqlConfiguration(new String[]{"none"},
+            org.elasticsearch.xpack.ql.util.DateUtils.UTC, "nobody", "cluster",
+            null, TimeValue.timeValueSeconds(30), -1, false, isCaseSensitive, "",
+            new TaskId("foobarbaz", 1234567890), () -> false);
+    }
```

Comment:
Can be removed and replaced with `EqlTestUtils.randomConfigurationWithCaseSensitive`.

Reply:
I tried that but ran into weird permission errors within the random utils when I tried that. I'll take another look to see

Reply:
I've seen that issue as well and, temporarily, I've passed `null` as the TaskId in the `TEST_CFG` to progress with my PR. @imotov might know more about why we have this behavior.

Reply:
Can either of you create an issue when you get the exception again. I'm not clear why a permission exception would be triggered and thus, if null, the workaround should be used inside the Utils class directly.